tests/integration/mirrorpeer_status_test.go

```diff
@@ -0,0 +1,191 @@
+		Name:      name,
+		Namespace: namespace,
+	}
+}
+
+var _ = Describe("MirrorPeer Status Tests", func() {
+
+	fakeReconciler, fakeMirrorPeer, err := GetFakeMirrorPeerReconciler()
+	Expect(err).NotTo(HaveOccurred())
+	err = fakeReconciler.Client.Create(context.TODO(), GetFakeManagedCluster(testCluster1), &client.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	err = fakeReconciler.Client.Create(context.TODO(), GetFakeManagedCluster(testCluster2), &client.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	err = fakeReconciler.Client.Create(context.TODO(), GetFakeNamespace(testCluster1), &client.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	err = fakeReconciler.Client.Create(context.TODO(), GetFakeNamespace(testCluster2), &client.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	When("Source secrets have not been created", func() {
+		It("should have Phase to be ExchangingSecret", func() {
+
+			res, err := fakeReconciler.Reconcile(context.TODO(), reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      fakeMirrorPeer.Name,
+					Namespace: fakeMirrorPeer.Namespace,
+				},
+			})
+			Expect(res.RequeueAfter).To(BeEquivalentTo(time.Second * 5))
+			Expect(err).NotTo(HaveOccurred())
+
+			var mp multiclusterv1alpha1.MirrorPeer
+			err = fakeReconciler.Client.Get(context.TODO(), types.NamespacedName{Name: fakeMirrorPeer.Name, Namespace: fakeMirrorPeer.Namespace}, &mp)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(mp.Status.Phase).To(BeEquivalentTo(multiclusterv1alpha1.ExchangingSecret))
+		})
+	})
+
+	When("Source secrets have been created", func() {
+		BeforeEach(func() {
+			pr1 := fakeMirrorPeer.Spec.Items[0]
+			pr2 := fakeMirrorPeer.Spec.Items[1]
+
+			secretNN1 := getNamespacedName(common.CreateUniqueSecretName(pr1.ClusterName, pr1.StorageClusterRef.Namespace, pr1.StorageClusterRef.Name), pr1.ClusterName)
+
+			storageClusterNN1 := getNamespacedName(pr1.StorageClusterRef.Name, pr1.StorageClusterRef.Namespace)
+
+			sec1 := common.CreateSourceSecret(secretNN1, storageClusterNN1, []byte("SecretData1"))
+
+			secretNN2 := getNamespacedName(common.CreateUniqueSecretName(pr2.ClusterName, pr2.StorageClusterRef.Namespace, pr2.StorageClusterRef.Name), pr2.ClusterName)
+			storageClusterNN2 := getNamespacedName(pr2.StorageClusterRef.Name, pr2.StorageClusterRef.Namespace)
+
+			sec2 := common.CreateSourceSecret(secretNN2, storageClusterNN2, []byte("SecretData2"))
+
+			err = fakeReconciler.Client.Create(context.TODO(), sec1, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = fakeReconciler.Client.Create(context.TODO(), sec2, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+
+		It("should have Phase to be ExchangedSecret", func() {
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      fakeMirrorPeer.Name,
+					Namespace: fakeMirrorPeer.Namespace,
+				},
+			}
+
+			_, err = fakeReconciler.Reconcile(context.TODO(), req)
```

Comment:
Each `It()` block starts with a reconcile. If we want, we could look into `JustBeforeEach()` block.

The usual pattern is, `BeforeEach()` block sets the common configurations and `JustBeforeEach()` block starts the main program (for us it is the `Reconcile` call). So the sequence will be something like this,
[Describe level] `BeforeEach()` -> [When level] `BeforeEach()` -> `JustBeforeEach()` -> `It()`

That means, in each `When()` block, as well, we can have custom `BeforeEach()` block which sets that particular test specific configuration.